response_operations_ui/views/accounts.py

```diff
@@ -104,66 +101,101 @@ def change_account_name():
+    notify_controller = NotifyController()
+
+    update_account_details_personalisation = {
+        "first_name": first_name,
+        "value_name": "email",
+        "changed_value": new_email,
+    }
+    token = token_decoder.generate_token(json.dumps({"email": new_email, "user_id": user_id}))
+    verification_url = f"{app.config['RESPONSE_OPERATIONS_UI_URL']}/account/verify-email/{token}"
+    update_email_personalisation = {"CONFIRM_EMAIL_URL": verification_url, "first_name": first_name}
+    logger.info("Sending update account verification email", verification_url=verification_url)
+
+    try:
+        notify_controller.request_to_notify(
+            email=new_email, template_name="update_email", personalisation=update_email_personalisation
+        )
+        notify_controller.request_to_notify(
+            email=current_email,
+            template_name="update_account_details",
+            personalisation=update_account_details_personalisation,
+        )
+    except NotifyError as e:
+        logger.error(
+            f"Error sending change of email acknowledgement to Notify Gateway for {user_id}", msg=e.description
+        )
+        return False
+
+    return True
+
+
+def _can_change_username_and_email(new_email: str) -> bool:
+    user_filter = f"email+eq+%22{new_email}%22+or+userName+eq+%22{new_email}%22"
+    uaa_user = uaa_controller.get_user_by_filter(user_filter)
+
+    if uaa_user and uaa_user["totalResults"] == 0:
+        return True
+
+    return False
+
+
 @account_bp.route("/verify-email/<token>", methods=["GET"])
-@login_required
 def verify_email(token):
     try:
         duration = app.config["UPDATE_ACCOUNT_EMAIL_TOKEN_EXPIRY"]
         json_token = token_decoder.decode_email_token(token, duration)
         token_dict = json.loads(json_token)
-        user_id = session["user_id"]
-        if token_dict["user_id"] == user_id:
-            user_from_uaa = uaa_controller.get_user_by_id(user_id)
-            user_from_uaa["emails"][0]["value"] = token_dict["email"]
-            logger.info("Updating email in UAA")
-            errors = uaa_controller.update_user_account(user_from_uaa)
-            if errors is not None:
-                logger.error("Error updating email in UAA", msg=errors["message"])
-                flash("Failed to update email. Please try again", category="warn")
-            else:
-                flash("Your email has been changed", category="successful_signout")
-                return redirect(url_for("account_bp.confirm_email_change"))
+
+        user_from_uaa = uaa_controller.get_user_by_id(token_dict["user_id"])
+
+        if user_from_uaa is None:
+            return render_template("request-new-account-error.html")
+
+        user_from_uaa["emails"][0]["value"] = token_dict["email"]
+        user_from_uaa["userName"] = token_dict["email"]
+
+        errors = uaa_controller.update_user_account(user_from_uaa)
```

Comment:
You could call _can_change_username_and_email however the likelihood of it happening are very slim and you increase the amount of calls. There is error catching here anyway if there is an issue, so decide to not to use it